rust/main/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -230,39 +222,39 @@ where
         &self,
         logs: Vec<(Indexed<T>, LogMeta)>,
         stored_logs_metric: &GenericCounter<AtomicU64>,
-    ) -> (Vec<(Indexed<T>, LogMeta)>, bool) {
+    ) -> Vec<(Indexed<T>, LogMeta)> {
         let deduped_logs = HashSet::<_>::from_iter(logs);
         let logs = Vec::from_iter(deduped_logs);
 
-        // Store deliveries
-        let stored_logs_count = match self.store.store_logs(&logs).await {
-            Ok(stored) => stored,
-            Err(err) => {
-                warn!(?err, "Error storing logs in db");
-                Default::default()
+        let deduped_logs_count = logs.len() as u64;
+        let mut accumulated_stored_logs_count: u64 = 0;
+
+        loop {
```

Comment:
nit: `loop` to me indicates a long-running process, but since this is only expected to do a couple of iterations until full success I'd just use a `while accumulated_stored_logs_count < deduped_logs_count`